src/studio/ui/StudioPanel.java

```diff
@@ -1979,8 +1997,50 @@ public void propertyChange(PropertyChangeEvent pce){
         dividerLastPosition=splitpane.getDividerLocation();
     }
 
+    private JPopupMenu createJPopupMenu() {
+        JPopupMenu popup = new JPopupMenu();
+        JMenuItem close = new JMenuItem("Close");
+        close.addActionListener(e -> {
+            int idx = tabbedPane.getSelectedIndex();
+            if (idx >= 0) {
+                tabbedPane.removeTabAt(idx);
+            }
+        });
+
+        JMenuItem closeOthers = new JMenuItem("Close others");
+        closeOthers.addActionListener(e -> {
+            Component selected = tabbedPane.getSelectedComponent();
+            if (selected != null) {
+                int selectedIndex = tabbedPane.getSelectedIndex();
+                if (selectedIndex > 0) {
+                    String title = tabbedPane.getTitleAt(selectedIndex);
+                    Icon icon = tabbedPane.getIconAt(selectedIndex);
+                    tabbedPane.setComponentAt(tabbedPane.getSelectedIndex(), null);
+                    tabbedPane.setComponentAt(0, selected);
+                    tabbedPane.setTitleAt(0, title);
+                    tabbedPane.setIconAt(0, icon);
+                }
+
+                while (tabbedPane.getTabCount() > 1) {
+                    tabbedPane.removeTabAt(1);
+                }
+
+                tabbedPane.setSelectedIndex(0);
+            }
+        });
+        JMenuItem closeAll = new JMenuItem("Close all");
+        closeAll.addActionListener(e -> tabbedPane.removeAll());
+
+        popup.add(close);
```

Comment:
In modern browsers, middle-click on a tab closes it. The same should work here.

Reply:
Good idea - will do in another PR if you don't mind

Reply:
https://github.com/finos/kdb-studio/pull/35